Lesson17/ecenaz/contactBook.js

```diff
@@ -0,0 +1,235 @@
+      );
+    }
+  }
+}
+
+console.log("Initial contact list:");
+displayAllContacts(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already 
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+
+function addContact(newName, newPhone, newEmail, contactList) {
+  for (let i = 0; i < contactList.length; i++) {
+    if (newName === contactList[i].name) {
+      console.warn("The contact already exist");
+      return;
+    }
+  }
+  const contactListLengthPrev = contactList.length;
+  contactList.push({
+    name: newName,
+    phone: newPhone,
+    email: newEmail,
+  });
+  const contactListLengthNew = contactList.length;
+
+  if (contactListLengthPrev < contactListLengthNew) {
+    console.log("Contact added successfully.");
+  } else {
+    console.log("An error occured when adding a contact.");
+  }
+
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator
+  // contactListLengthPrev < contactListLengthNew
+  //   ? console.log("Contact added successfully.")
+  //   : console.log("An error occured when adding a contact.");
+}
+
+console.log("\nAdding a new contact: Charlie");
+addContact("Charlie", "777-777-7777", "charlie@example.com", contacts);
+displayAllContacts(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function viewContact(name, contactList) {
+  let isContactFound = false;
+  for (const contact of contactList) {
+    if (name === contact.name) {
+      console.log(
+        `name: ${contact.name}, phone:${contact.phone}, email:${contact.email}`,
+      );
+      isContactFound = true;
+    }
+  }
+  if (!isContactFound) {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+console.log("\nViewing Bob's contact:");
+viewContact("Bob", contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Contact
+-----------------------------------------------------------
+Function: updateContact(name, newPhone, newEmail)
+- Finds the contact by name and updates phone + email.
+- Logs "Contact updated successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function updateContact(name, newPhone, newEmail, contactList, findContact) {
+  const isContactFound = findContact(contacts, contactList);
```

Comment:
The goal of `findContact` is to find a contact in the **contactList** by contact's **name**. Therefore, we pass `name` and `contactList` as parameters. 
```suggestion
  const isContactFound = findContact(name, contactList);
```